packages/filebrowser/src/listing.ts

```diff
@@ -860,6 +882,22 @@ class DirListing extends Widget {
     default:
       break;
     }
+
+    // Detects printable characters typed by the user.
+    // Not all browsers support .key, but it discharges us from reconstructing
+    // characters from key codes.
+    if (event.key !== undefined && event.key.length === 1) {
```

Comment:
We can import `getKeyboardLayout` from `@phosphor/keyboard` and use `.keyForKeydownEvent` to get the correct value.  https://github.com/phosphorjs/phosphor/blob/297d7d247d69de6d329336ab6e30fd26bd4c204f/packages/keyboard/src/index.ts#L50